pkg/handlers/generic/lifecycle/csi/handler.go

```diff
@@ -74,68 +74,103 @@ func (c *CSIHandler) AfterControlPlaneInitialized(
 	)
 	varMap := variables.ClusterVariablesToVariablesMap(req.Cluster.Spec.Topology.Variables)
 	resp.SetStatus(runtimehooksv1.ResponseStatusSuccess)
-	csiProviders, err := variables.Get[v1alpha1.CSI](
+	csi, err := variables.Get[v1alpha1.CSI](
 		varMap,
 		c.variableName,
 		c.variablePath...)
 	if err != nil {
-		if variables.IsNotFoundError(err) ||
-			csiProviders.Providers == nil ||
-			len(csiProviders.Providers) == 0 {
-			log.V(4).Info(
-				fmt.Sprintf(
-					"Skipping CSI handler, no providers given in %v",
-					csiProviders,
-				),
-			)
+		if variables.IsNotFoundError(err) {
+			log.Info("Skipping CSI handler, the cluster does not define the CSI variable")
 			return
 		}
 		log.Error(
 			err,
-			"failed to read CSI provider from cluster definition",
+			"failed to read the CSI variable from the cluster",
 		)
 		resp.SetStatus(runtimehooksv1.ResponseStatusFailure)
 		resp.SetMessage(
-			fmt.Sprintf("failed to read CSI provider from cluster definition: %v",
+			fmt.Sprintf("failed to read the CSI variable from the cluster: %v",
 				err,
 			),
 		)
 		return
 	}
-	if len(csiProviders.Providers) == 1 &&
-		len(csiProviders.Providers[0].StorageClassConfig) == 1 &&
-		csiProviders.DefaultStorage == nil {
-		csiProviders.DefaultStorage = &v1alpha1.DefaultStorage{
-			ProviderName:           csiProviders.Providers[0].Name,
-			StorageClassConfigName: csiProviders.Providers[0].StorageClassConfig[0].Name,
+
+	if len(csi.Providers) == 0 {
```

Comment:
This can be added as `minItems` in the API schema can't it? I don't think we need custom logic for this. Happy to leave this here as well as defensive programming but let's also update the API to prevent bad data getting in.

Reply:
Yes, I meant to add that, but got distracted by the test :smile:

Reply:
Let's use validation and keep this.